src/py/pyodide/_package_loader.py

```diff
@@ -149,6 +155,25 @@ def should_load_dynlib(path: str):
     return not PLATFORM_TAG_REGEX.match(tag)
 
 
+def set_wheel_installer(
+    filename: str,
+    archive: IO[bytes],
+    target_dir: Path,
+    installer: str | None,
+    source: str | None,
+):
+    z = ZipFile(archive)
+    name, version = filename.split("-", 2)[:-1]
+    dist_info_name = f"{name}-{version}.dist-info"
+    if not (zipfile.Path(z) / dist_info_name).is_dir():
+        raise Exception("archive is not a valid wheel")
```

Comment:
Maybe it would be good to print the name of the archive here, otherwise, it might be hard to figure out from the traceback which package failed